client/src/components/Form/Elements/FormRulesEdit.vue

```diff
@@ -0,0 +1,96 @@
+<script setup>
+import RuleCollectionBuilder from "components/RuleCollectionBuilder";
+import RulesDisplay from "components/RulesDisplay/RulesDisplay";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { ref, computed } from "vue";
+import { getAppRoot } from "onload/loadConfig";
+import axios from "axios";
+
+const props = defineProps({
+    value: {
+        type: Object,
+    },
+    target: {
+        type: Object,
+        default: null,
+    },
+});
+
+const modal = ref(null);
+const elements = ref(null);
+
+const initialRules = {
+    rules: [],
+    mapping: [],
+};
+
+const displayRules = computed(() => props.value ?? initialRules);
+
+async function onEdit() {
+    if (props.target) {
+        const url = `${getAppRoot()}api/dataset_collections/${props.target.id}?instance_type=history`;
```

Comment:
I know this is from the previous code, but we could use this opportunity to use `UrlData` from `utils/url` here.